lib/pages/class_feedback/view/class_feedback_view.dart

```diff
@@ -0,0 +1,404 @@
+        .fetchQuestions()
+        .then((questions) => setState(() => feedbackQuestions = questions));
+
+    for (int i = 0; i <= feedbackQuestions.length; i++) {
+      initialValues.insert(i, {
+        0: false,
+        1: false,
+        2: false,
+        3: false,
+        4: false,
+      });
+    }
+
+    return feedbackQuestions;
+  }
+
+  Widget classWidget() {
+    return TextFormField(
+      enabled: false,
+      controller: classController,
+      decoration: InputDecoration(
+        labelText: S.current.labelClass,
+        prefixIcon: const Icon(FeatherIcons.bookOpen),
+      ),
+      onChanged: (_) => setState(() {}),
+    );
+  }
+
+  Widget lecturerWidget(BuildContext context) {
+    final personProvider = Provider.of<PersonProvider>(context);
+
+    return FutureBuilder(
+      future: personProvider.mostRecentLecturer(widget.classHeader.id),
+      builder: (context, snapshot) {
+        if (snapshot.connectionState == ConnectionState.done) {
+          final lecturerName = snapshot.data;
+          selectedTeacherName = lecturerName;
+          return TextFormField(
+            enabled: false,
+            controller: TextEditingController(text: lecturerName ?? '-'),
+            decoration: InputDecoration(
+              labelText: S.current.labelLecturer,
+              prefixIcon: const Icon(Icons.person_outline),
+            ),
+            onChanged: (_) => setState(() {}),
+          );
+        } else {
+          return const Center(child: CircularProgressIndicator());
+        }
+      },
+    );
+  }
+
+  Widget assistantWidget() {
+    return AutocompletePerson(
+      key: const Key('AutocompleteAssistant'),
+      labelText: S.current.labelAssistant,
+      warning: S.current.warningYouNeedToSelectAssistant,
+      formKey: formKey,
+      onSaved: (value) => selectedAssistant = value,
+      classTeachers: classTeachers,
+    );
+  }
+
+  Widget acknowledgementWidget() {
+    return Padding(
+      padding: const EdgeInsets.all(10),
+      child: Row(
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: [
+          Checkbox(
+            value: agreedToResponsibilities,
+            visualDensity: VisualDensity.compact,
+            onChanged: (value) =>
+                setState(() => agreedToResponsibilities = value),
+          ),
+          Expanded(
+            child: Padding(
+              padding: const EdgeInsets.only(top: 10.25),
+              child: Text(
+                S.current.messageAgreeFeedbackPolicy,
+                style: Theme.of(context).textTheme.subtitle1,
+              ),
+            ),
+          ),
+        ],
+      ),
+    );
+  }
+
+  Widget categoryHeader(String category) {
+    return Column(
+      children: [
+        Text(
+          category,
+          style: Theme.of(context).textTheme.headline6,
+        ),
+        const SizedBox(height: 24),
+      ],
+    );
+  }
+
+  Widget questionWidget(FeedbackQuestion question) {
```

Comment:
Make this an actual, separate widget (it can have separate methods for each questions, so it's easier to organise the code).